clippy_lints/src/non_copy_const.rs

```diff
@@ -128,7 +128,7 @@ fn verify_ty_bound<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, ty: Ty<'tcx>, source: S
                 db.span_label(const_kw_span, "make this a static item (maybe with lazy_static)");
             },
             Source::Assoc { ty: ty_span, .. } => {
-                if ty.flags.contains(TypeFlags::HAS_FREE_LOCAL_NAMES) {
+                if !ty.has_infer_types_or_consts() {
```

Comment:
I think that this should keep the meaning of the original code:
```suggestion
                if ty.flags.intersects(TypeFlags::HAS_FREE_LOCAL_NAMES) {
```

Reply:
Ah, actually it broke current suggestions, thanks!